Sources/Amplitude/Utilities/QueueTimer.swift

```diff
@@ -13,27 +13,25 @@ internal class QueueTimer {
         case resumed
     }
 
-    let interval: TimeInterval
     let timer: DispatchSourceTimer
-    let queue: DispatchQueue
     let handler: () -> Void
 
     @Atomic var state: State = .suspended
 
-    static var timers = [QueueTimer]()
-
-    static func schedule(interval: TimeInterval, queue: DispatchQueue = .main, handler: @escaping () -> Void) {
-        let timer = QueueTimer(interval: interval, queue: queue, handler: handler)
-        Self.timers.append(timer)
+    /// `DispatchQueue` appropriate for the environment being executed.
+    ///
+    /// When running on a server (in a headless/windowless state), the `.main` `DispatchQueue`
+    /// may not execute tasks that are enqueued to it. Essentially we can't count on the CGD event loop
+    /// like we typically do in iOS/tvOS/watchOS/macOS applications.
+    static func queue(runningOnServer: Bool) -> DispatchQueue {
+        runningOnServer ? .global(qos: .utility) : .main
     }
 
-    init(interval: TimeInterval, once: Bool = false, queue: DispatchQueue = .main, handler: @escaping () -> Void) {
-        self.interval = interval
-        self.queue = queue
+    init(interval: TimeInterval, once: Bool = false, runningOnServer: Bool = false, handler: @escaping () -> Void) {
```

Comment:
Could we revert the change in `QueueTimer`? As it is supposed to be a general reusable component. If we want to go with this `runningOnServer` solution, we can update set the queue in these places based on `configuration`:
- https://github.com/amplitude/Amplitude-Swift/blob/0aea976ab33e5ff2f0e555cad7406ca4773422b1/Sources/Amplitude/Utilities/EventPipeline.swift#L30
- https://github.com/amplitude/Amplitude-Swift/blob/0aea976ab33e5ff2f0e555cad7406ca4773422b1/Sources/Amplitude/Utilities/IdentifyInterceptor.swift#L193